File: backends/ruby-hanami/app/domain/notification/value_objects.rb
# frozen_string_literal: true
#
# 通知の値オブジェクトを定義するドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Notification
    # 通知ページネーション結果の値オブジェクトです。
    Page = Struct.new(:items, :next_cursor, keyword_init: true)

    # 通知件数の値オブジェクトです。
    CountsVo = Struct.new(:unread, :total, keyword_init: true)
  end
end

File: backends/kotlin-ktor/src/main/kotlin/com/authorization/domain/notification/ValueObjects.kt
/*
 * 通知ドメインの値オブジェクトモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.domain.notification

/**
 * 通知ページの値オブジェクトです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class Page(
    val items:      List<Notification>,
    val nextCursor: String?,
)

/**
 * 通知件数の値オブジェクトです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class CountsVo(
    val unread: Long,
    val total:  Long,
)
